bazel/envoy_build_system.bzl

```diff
@@ -0,0 +1,41 @@
+# References to Envoy external dependencies should be wrapped with this function.
+def envoy_external_dep_path(dep):
+    return "//external:%s" % dep
+
+ENVOY_COPTS = [
+    "-ggdb3", "-fno-omit-frame-pointer", "-Wall", "-Wextra", "-Werror",
+    "-Wnon-virtual-dtor", "-Woverloaded-virtual", "-Wold-style-cast",
+    "-std=c++0x", "-includesource/precompiled/precompiled.h",
+    "-isystem", "include", "-isystem", "source"
+]
+
+ENVOY_LINKOPTS = ["-static-libstdc++", "-static-libgcc"]
```

Comment:
Yeah, I originally was thinking about envoy_cc_binary, but we're aways from that.